### ISSUE TITLE
Report mode

### ISSUE THREAD

**Issue by jtyberg** [jupyter-incubator/dashboards#55]
I have a few lengthy notebooks, each with a large number of cells, and I want to surface more cells than would fit in a single screen dashboard view.  So what I want to show is more like a report than a dashboard.  

Say out of 100 cells, I want to select 20 to show up when I switch to View Dashboard.  I don't really want to layout the cells, I just want to display them top to bottom, in the same order they show up in Notebook mode.

Ideally, I would be able to select which cells to display while in Notebook mode, by clicking something (icon?) in the top right corner of the cell.  Then I could switch to View Dashboard mode to see the result.  Layout Dashboard would not be necessary in this case.

**Comment by parente** [comment]
#56 does help on this one a bit, but we don't know how to get a better indicator in notebook view of what's in the layout and where.

**Comment by jtyberg** [comment]
I'm not giving up on this.  Long before I decide on what few widgets/cells I want to show in a dashboard, I'm going back and forth with folks, sharing notebooks in what I call Report mode.  With a few cells in these notebooks (OK, more than 50), choosing ones to show and laying them out is a mental exercise that I do not care to embark on.  What's more, I like to just share the whole dang notebook when I'm sharing with multiple people, because what happens while we're on the phone talking about it, is that different people focus on one section or the other.

Currently, to simulate a report style notebook, there are issues.  

* First, I have to go into Layout mode and select every cell to be shown.  Selecting and adding multiple cells at once (#130) would help here.
* There is no good way to navigate in View mode.  I don't want to write a bunch of markup/HTML/links to create a table of contents, nor do I think it would be useful to do so.  I also don't want to add 1.x, 2.x, ... in section headers in my notebook because it makes my head hurt, and again, I don't think it would be useful in helping multiple people navigate around a notebook together.  When you're looking at a plot, you'd have to scroll way up to a section header or whatever to know if you're in the right spot, then what, scroll back down?  What I DO think would be useful is adding cell numbers (they can be subtle) to every cell that is shown in View mode.  The cells can be numbered in order left-to-right, top-to-bottom.  What I've been doing is converting my whole notebook to HTML, input code cells and all, after running the notebook cells top-to-bottom, in order.  This gives me cell numbers on every cell, and guess what, **it works**.  John: "I'm looking at cell 17.  How is it possible to get a negative value for this metric?"  Jane: "In the plot in cell 6, what is the unit on the x-axis?"

I'm not sure what to do about the cell numbers for navigation.  I'm hesitant to open an issue because I'm blurring the line between dashboard and report styles of presenting a notebook.

**Comment by dalogsdon** [comment]
@jtyberg:
> I like to just share the whole dang notebook when I'm sharing with multiple people
...
> First, I have to go into Layout mode and select every cell to be shown. Selecting and adding multiple cells at once (#130) would help here.

Does the `Cell > Dashboard > Show All Cells` help in this case, assuming  you mean to share every cell in the notebook?

**Comment by parente** [comment]
> Does the Cell > Dashboard > Show All Cells help in this case, assuming you mean to share every cell in the notebook?

If it changes to laying them out simply top to bottom, full width, then I assume yes. But right now, it does that dense grid packing. I think we should revisit that decision and simplify.

**Comment by dalogsdon** [comment]
@jtyberg I think what you are getting at is a use case where Dynamic Dashboards is not the best tool for the job of collaborating on a notebook, since you are sharing the entire notebook and scrolling the page. I agree that numbering the cells is helpful. In my experience, a dashboard is a single-page entity (perhaps with tabs but no scrolling) meant to show a few quick pieces of information. You have titled this issue "Report Mode," so perhaps this functionality is meant for a different notebook extension and doesn't belong in dashboards?

cc @parente @jhpedemonte

**Comment by parente** [comment]
Spoke with @jtyberg about his use case. There's a bigger picture UX that spans a few issues here (#117, #130). I'm going to open a new issue to tackle just the one "report mode" piece of that big picture and try to re-couch it in terms of dashboards.

**Comment by jtyberg** [comment]
I like to think of using a notebook in terms of phases:

1.  There's the exploratory phase where I've got cells everywhere, I'm poking at everything, and in general, making a big mess.  
2. Then, once I've got my feet under me, I start to clean up a bit.  I remove all the gorp cells, and begin to tell a story within the notebook.  I don't know yet what is going to be useful or insightful.  I need the help of other stakeholders to be sure.  This is when I need to just share the whole notebook, in a somewhat cleaned up rendering (e.g., no code showing, etc).
3. Further down the line, I know what bits of the notebook are dashboard worthy.  Now I can select the 5 cells that should be included in a single page view and hosted somewhere.

I agree that dashboards is really targeting number 3 in this list.  But 1 and 2 still exist, and they are necessary to get to 3.  The promise of dashboards is that the notebook user does not have to leave the notebook and go write a separate web app to encapsulate what's in it.  So I really don't want to go outside the notebook before I'm dashboard ready either.

Hope that makes sense.
